src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,29 @@
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
+        }
+        Transaction transaction = session.beginTransaction();
+        try {
+            session.persist(country);
```

Comment:
You don't need to set an ID for your entity explicitly as Hibernate will do it for you if the mapping is set up correctly. Make sure your entity class `Country` is correctly annotated with `@GeneratedValue` for the ID field.